src/sentry/models/projectkey.py

```diff
@@ -229,11 +224,18 @@ def js_sdk_loader_cdn_url(self):
                 reverse("sentry-js-sdk-loader", args=[self.public_key, ".min"]),
             )
 
-    def get_endpoint(self):
-        endpoint = settings.SENTRY_PUBLIC_ENDPOINT or settings.SENTRY_ENDPOINT
-        if not endpoint:
-            endpoint = options.get("system.url-prefix")
-        return endpoint
+    def get_endpoint(self, public=True):
+        if public:
+            endpoint = settings.SENTRY_PUBLIC_ENDPOINT or settings.SENTRY_ENDPOINT
+        else:
+            endpoint = settings.SENTRY_ENDPOINT
+
+        if endpoint:
+            return endpoint.format(
+                organization_id=self.project.organization_id, project_id=self.project_id
```

Comment:
Did we also want to allow formatting `project_id`? I was under the assumption it would only be the organization, since the project ID is already in the path.